cmd/manager/operator.go

```diff
@@ -187,73 +197,125 @@ func RunOperator(cmd *cobra.Command, args []string) {
 	// Get a config to talk to the apiserver
 	cfg, err := config.GetConfig()
 	if err != nil {
-		log.Error(err, "")
+		setupLog.Error(err, "")
 		os.Exit(1)
 	}
 
 	ctx := context.TODO()
-	// Become the leader before proceeding
-	err = leader.Become(ctx, "compliance-operator-lock")
+	kubeClient := kubernetes.NewForConfigOrDie(cfg)
+	monitoringClient := monclientv1.NewForConfigOrDie(cfg)
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                 operatorScheme,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "81473831.openshift.io", // operator-sdk generated this for us
+	})
 	if err != nil {
-		log.Error(err, "")
+		setupLog.Error(err, "unable to create manager")
 		os.Exit(1)
 	}
 
-	kubeClient := kubernetes.NewForConfigOrDie(cfg)
-	monitoringClient := monclientv1.NewForConfigOrDie(cfg)
+	setupLog.Info("Registering Components.")
 
-	// Create a new Cmd to provide shared dependencies and start components
-	mgr, err := manager.New(cfg, options)
-	if err != nil {
-		log.Error(err, "")
+	if err = (&scanctrl.ReconcileComplianceScan{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ComplianceScan")
+		os.Exit(1)
+	}
+	if err = (&remctrl.ReconcileComplianceRemediation{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ComplianceRemediation")
+		os.Exit(1)
+	}
+	if err = (&suitectrl.ReconcileComplianceSuite{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ComplianceSuite")
+		os.Exit(1)
+	}
+	if err = (&profilectrl.ReconcileProfileBundle{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ProfileBundle")
+		os.Exit(1)
+	}
+	if err = (&ssbctrl.ReconcileScanSettingBinding{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
```

Comment:
Since this isn't passing in a roleVal it causes the operator to go into a panic when it attempts to reconcile the scan setting binding.

Reply:
Did some more debugging here:

```
Reconciler in SetupWithManager: &{0xc0008c34a0 0xc000446e00 <nil> <nil> <nil> <nil>}
Reconciler in newReconciler: &{0xc0008c34a0 0xc000446e00 0xc0004f7290 0xc0005e1920 0xc00032a0a0 0xc00032a280}
Reconciler in validateRoles: &{0xc0008c34a0 0xc000446e00 <nil> <nil> <nil> <nil>}
```

I think this is causing the panic issues since it's using *this* instance of ReconcileScanSettingBinding at run time and not the one created by `newReconciler`, which contains the default values for `roleVal`.